### PR TITLE
address the CommonJS module system

### PR DESCRIPTION
**Pull request by gomain**
I found the this part of the text (about interoperating with JavaScript module) hard to follow. I try to improve it in these ways
  - Naturally say that what is _imported_  in this module is _exported_ from some other module. In this case, it's from a JavaScript module.
  - Make explicit that it's the _CommonJS_ module system that is used. I personally don't have experience about JavaScript module systems and needed the day to research. This version should put an interested reader in right direction. And is a bare minimum to keep going.
  - Mention the same-name principle.
  - Remove part about Spago's search path & compiler behavior.
  - Remove part about generated code optimization.
  - Some minor choice of words.

Removed parts probably belong to some other part of the book. But here it seems non-relevant.

### CODE REVIEW

File: text/chapter10.md [hunk 1]
```diff
@@ -57,17 +57,15 @@ We can assign this type to the function with the following foreign import declar
 {{#include ../exercises/chapter10/test/URI.purs}}
 ```
 
-We also need to write a foreign JavaScript module. If the module above is saved as `test/URI.purs`, then the foreign JavaScript module should be saved as `test/URI.js`:
+We also need to write a foreign JavaScript module to import it from. Purescript uses the CommonJS module system when interoperating with JavaScript. A corresponding foreign JavaScript module is one of the same name but extension changed from `.purs` to `.js`. If the Purescript module above is saved as `URI.purs`, then the foreign JavaScript module is saved as `URI.js`:
```
Comment (milesfrain) [inline text/chapter10.md:60]: Might be good to mention that we're planning to convert to ES modules for the next compiler release (`0.15`).
https://github.com/purescript/purescript/pull/3791

**Comment by milesfrain** [comment]
Looks good. Thoughts on whether to mention the upcoming ES Modules change? That note might help ensure that future readers aren't confused if we forgot to update that section later.

File: text/chapter10.md (hunk 1 above)
Comment (gomain) [reply text/chapter10.md:60]: Probably not. I don't think the "book" should account for unreleased features. When the time comes, we update the book.
